Tests/RabbitMq/ConsumerTest.php

```diff
@@ -341,4 +355,121 @@ public function testGracefulMaxExecutionWontWaitIfPastTheTimeout()
 
         $consumer->consume(1);
     }
+
+    public function testTimeoutWait()
+    {
+        // set up amqp connection
+        $amqpConnection = $this->prepareAMQPConnection();
+        // set up amqp channel
+        $amqpChannel = $this->prepareAMQPChannel();
+        $amqpChannel->expects($this->atLeastOnce())
+            ->method('getChannelId')
+            ->with()
+            ->willReturn(true);
+        $amqpChannel->expects($this->once())
+            ->method('basic_consume')
+            ->withAnyParameters()
+            ->willReturn(true);
+
+        // set up consumer
+        $consumer = $this->getConsumer($amqpConnection, $amqpChannel);
+        // disable autosetup fabric so we do not mock more objects
+        $consumer->disableAutoSetupFabric();
+        $consumer->setChannel($amqpChannel);
+        $consumer->setTimeoutWait(30);
+        $consumer->setGracefulMaxExecutionDateTimeFromSecondsInTheFuture(60);
+        $consumer->setIdleTimeout(50);
+
+        $amqpChannel->callbacks = array('timeout_wait_test');
+
+        $amqpChannel->expects($this->exactly(2))
+            ->method('wait')
+            ->with(null, false, 30)
+            ->willReturnCallback(function ($allowedMethods, $nonBlocking, $waitTimeout) use ($consumer) {
+                // ensure max execution date time "counts down"
+                $consumer->setGracefulMaxExecutionDateTime(
+                    $consumer->getGracefulMaxExecutionDateTime()->modify("-$waitTimeout seconds")
+                );
+                // ensure last activity just occurred so idle timeout is not reached
+                $consumer->setLastActivityDateTime(new \DateTime());
+                throw new AMQPTimeoutException();
+            });
+
+        $consumer->consume(1);
+    }
+
+    public function testTimeoutWaitWontWaitPastGracefulMaxExecutionTimeout()
```

Comment:
Please rerun this test and check the failing reason:
`2) OldSound\RabbitMqBundle\Tests\RabbitMq\ConsumerTest::testTimeoutWaitWontWaitPastGracefulMaxExecutionTimeout
Expectation failed for method name is "wait" when invoked 1 time(s)
Parameter 2 for invocation PhpAmqpLib\Channel\AbstractChannel::wait(null, false, 9) does not match expected value.
Failed asserting that 9 matches expected 10.
/home/travis/build/eMAGTechLabs/RabbitMqBundle/RabbitMq/Consumer.php:95
/home/travis/build/eMAGTechLabs/RabbitMqBundle/Tests/RabbitMq/ConsumerTest.php:435`

Reply:
Check also:
`4) OldSound\RabbitMqBundle\Tests\RabbitMq\DynamicConsumerTest::testTimeoutWaitWontWaitPastGracefulMaxExecutionTimeout
Expectation failed for method name is "wait" when invoked 1 time(s)
Parameter 2 for invocation PhpAmqpLib\Channel\AbstractChannel::wait(null, false, 9) does not match expected value.
Failed asserting that 9 matches expected 10.
/home/travis/build/eMAGTechLabs/RabbitMqBundle/RabbitMq/Consumer.php:95
/home/travis/build/eMAGTechLabs/RabbitMqBundle/Tests/RabbitMq/ConsumerTest.php:435`